File: ruby/Loot.rb
#encoding:utf-8

require_relative 'LootToUI'

module Deepspace
    class Loot
        def initialize(sup, wea, shi, han, med, ef=false, city=false) #seran enteros, menos los parametros nuevos de la P4
            @nSupplies = sup
            @nWeapons = wea
            @nShields = shi
            @nHangars = han
            @nMedals = med
            @getEfficient = ef
            @spaceCity = city
        end

        def nSupplies
            @nSupplies
        end

        def nWeapons
            @nWeapons
        end

        def nShields
            @nShields
        end

        def nMedals
            @nMedals
        end

        def nHangars
            @nHangars
        end

        def efficient
            return @getEfficient
        end
        
        def spaceCity
            return @spaceCity
        end

        def to_s
            result = "NSupplies: #{@nSupplies}, nWeapons: #{@nWeapons}, nShields: #{@nShields}, nHangars: #{@nHangars}, nMedals: #{@nMedals}"
            result += ", getEfficient: " + efficient.to_s + ", spaceCity: " + spaceCity.to_s
            return result
        end

        def getUIversion
            return LootToUI.new(self)
        end
    end
end

File: java/Deepspace/src/deepspace/Loot.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author alvaro
 */
public class Loot {     //Se quita el public?. En los consultores se debe poner final?
    private int nSupplies;
    private int nWeapons;
    private int nShields;
    private int nHangars;
    private int nMedals;
    private boolean getEfficient;
    private boolean spaceCity;
    
    Loot(int nsu, int nw, int nsh, int nh, int nm){
        nSupplies = nsu;
        nWeapons = nw;
        nShields = nsh;
        nHangars = nh;
        nMedals = nm;
        getEfficient = false;
        spaceCity = false;
    }
    
    Loot(int nsu,int nw,int nsh, int nh,int nm, boolean ge, boolean sc){
        nSupplies = nsu;
        nWeapons = nw;
        nShields = nsh;
        nHangars = nh;   
        nMedals = nm;
        getEfficient = ge;
        spaceCity = sc;
    }
    
    public final boolean getEfficient(){
        return getEfficient;
    }
    
    public final boolean spaceCity(){
        return spaceCity;
    }
    
    public int getNSupplies(){
        return nSupplies;
    }
    
    public int getNWeapons(){
        return nWeapons;
    }
    
    public int getNShields(){
        return nShields;
    }
    
    public int getNHangars(){
        return nHangars;
    }
    
    public int getNMedals(){
        return nMedals;
    }
    
    LootToUI getUIversion(){
        return new LootToUI(this);
    }
    
    //EXTRA: PARA VER FACILMENTE EL CONTENIDO DE UNA INSTANCIA DE LOOT: (para la p2 lo pide, le cambio el nombre a toString)
    @Override
    public String toString(){
        String result = "Loot: ";
        result  = result + "NSupplies: " + getNSupplies() + "\n";
        result  = result + "    NWeapons: " + getNWeapons() + "\n";
        result  = result + "    NShields: " + getNShields() + "\n";
        result  = result + "    NHangars: " + getNHangars() + "\n";
        result  = result + "    NMedals: " + getNMedals() + "\n";
        
        return result;
    }
    
    
}
